### DISCUSSION TITLE
RHF | MUI | Controller | ImageUpload

### DISCUSSION THREAD

**Discussion by Nattyfro** [react-hook-form/react-hook-form#9890] [Q&A]
Dear RHF Support, 

I have created a form using RHF's FormProvider.api and useFormContext.api, which is spread across multiple components for readability. I have also built a feature where data is shown in a card as a separate component when a form field is entered, which is enabled through useWatch.api.

 I have encased a basic <input {....} /> with a <Controller/> to get the user to upload a profile picture using the Yup.js library for validation. However, when the delete button is selected the page goes blank, instead of reverting the component back to the 'upload image' state. I believe this is due to useCallback, and I would like to fix this issue.

 I have attached a CSB and the relevant video below for your reference.

Thank you for your time,

 Myles Y.

 
[https://codesandbox.io/s/avataruploadissue-4rlleg]
https://user-images.githubusercontent.com/89163481/217466455-5f59bf33-319e-47ba-9f89-87b55c9c7553.mp4

**Comment by leapful** [comment]
@Nattyfro 

No, it's not due to `useCallback`. The root cause is because you already altered the `value` attribute of HTML element `<input type="file" />` with `field.value` prop from `useController()` hook. The prop `field.value` is now a custom Date URL base64 to store your image, it cannot be passed to `<input type="file" />` as the input element only accepts file path.

Just exclude the `field.value` in case you are handling custom value with `<input type="file" />` to fix the problem.

Working demo: https://codesandbox.io/s/avataruploadissue-forked-dv9rlc?file=/src/AvatarUpload.tsx
